plugins/org.jkiss.dbeaver.data.transfer.ui/src/org/jkiss/dbeaver/tools/transfer/ui/pages/database/DatabaseConsumerPageMapping.java

```diff
@@ -1142,10 +1142,17 @@ private DatabaseMappingContainer[] getSelectedMappingContainers() {
 
     @Override
     public void activatePage() {
+        if (getWizard().getContainer() != null) {
+            // The user is importing something into a known container which is already loaded
+            loadSettings(true);
+            return;
+        }
+
         final DBPDataSourceContainer container = DatabaseConsumerSettings.getDataSourceContainer(getWizard().getSettings());
         final DBPPreferenceStore preferences = DTActivator.getDefault().getPreferences();
 
         if (preferences.getBoolean(DTConstants.PREF_RECONNECT_TO_LAST_DATABASE) || container != null && container.isConnected()) {
+            // The user is importing something into an unspecified container - reconnect to the last chosen container if the preference is set
             loadSettings(true);
         }
```

Comment:
I'd simplify it, it looks confusing for me now.

```suggestion
        if (getWizard().getContainer() != null ||
            DTActivator.getDefault().getPreferences().getBoolean(DTConstants.PREF_RECONNECT_TO_LAST_DATABASE) ||
            container != null && DatabaseConsumerSettings.getDataSourceContainer(getWizard().getSettings()).isConnected()) {
            // The user is importing something into an unspecified container - reconnect to the last chosen container if the preference is set
            loadSettings(true);
        }
```

Reply:
These are two separate cases, so I separated them and added comments.

Reply:
You may add comments to the if branches

Reply:
I'm okay if you did it intentionally, just don't like it :)